packages/java-packages/snippet-replacer-maven-plugin/src/main/java/azuresdk/plugin/SnippetReplacer.java

```diff
@@ -0,0 +1,383 @@
+        HashMap<String, List<String>> foundSnippets = new HashMap<String, List<String>>();
+
+        // scan the sample files for all the snippet files
+        for(Path samplePath: snippetSources){
+            List<String> sourceLines = Files.readAllLines(samplePath, StandardCharsets.UTF_8);
+            foundSnippets.putAll(this.GrepSnippets(sourceLines));
+        };
+
+        // walk across all the java files, run UpdateSrcSnippets
+        for(Path sourcePath: allLocatedJavaFiles){
+            discoveredIssues.addAll(this.VerifySrcSnippets(sourcePath, foundSnippets));
+        }
+
+        // now find folderToVerify/README.md
+        // run Update ReadmeSnippets on that
+        File readmeInBaseDir = new File(folderToVerify, "README.md");
+        discoveredIssues.addAll(this.VerifyReadmeSnippets(readmeInBaseDir.toPath(), foundSnippets));
+
+        if(discoveredIssues.size() > 0){
+            for(VerifyResult result: discoveredIssues){
+                System.out.println(String.format("SnippetId %s needs update in file %s.", result.SnippetWithIssues, Files.readString(result.ReadmeLocation)));
+            }
+            throw new MojoExecutionException("Discovered snippets in need of updating. Please run this plugin in update mode and commit the changes.");
+        }
+    }
+
+    public void RunUpdate(File folderToVerify) throws IOException{
+        List<Path> allLocatedJavaFiles = _glob(folderToVerify.toPath(), JAVA_GLOB);
+        List<Path> snippetSources = _globFiles(allLocatedJavaFiles, SAMPLE_PATH_GLOB);
+        HashMap<String, List<String>> foundSnippets = new HashMap<String, List<String>>();
+
+        // scan the sample files for all the snippet files
+        for(Path samplePath: snippetSources){
+            List<String> sourceLines = Files.readAllLines(samplePath, StandardCharsets.UTF_8);
+            foundSnippets.putAll(this.GrepSnippets(sourceLines));
+        };
+
+        // walk across all the java files, run UpdateSrcSnippets
+        for(Path sourcePath: allLocatedJavaFiles){
+            this.UpdateSrcSnippets(sourcePath, foundSnippets);
+        }
+
+        // now find folderToVerify/README.md
+        // run Update ReadmeSnippets on that
+        File readmeInBaseDir = new File(folderToVerify, "README.md");
+        this.UpdateReadmeSnippets(readmeInBaseDir.toPath(), foundSnippets);
+    }
+
+    public void UpdateReadmeSnippets(Path file, HashMap<String, List<String>> snippetMap) throws IOException {
+        List<String> lines = Files.readAllLines(file, StandardCharsets.UTF_8);
+        StringBuilder modifiedLines = this.UpdateSnippets(lines, snippetMap, "```Java", "```", 0, "");
+
+        if(modifiedLines != null) {
+            try (FileWriter modificationWriter = new FileWriter(file.toAbsolutePath().toString(), StandardCharsets.UTF_8)) {
+                modificationWriter.write(modifiedLines.toString());
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
Instead of swallowing the exception, this should fail the maven build. Same for other places where an exception is caught.

Reply:
Actually totally agree actually. My intent was to throw the IOException all the way up to the entry point which should print the stack trace and exit. Old code that needs cleaning.